sheetshuttle/util.py

```diff
@@ -78,4 +78,9 @@ def fill_to_dimensions(
         for _ in range(0, row_difference):
             data.append([None] * columns)
 
+    for list in data:
+        for i in range(len(list)):
+            if list[i] == "":
+                list[i] = None
+
```

Comment:
I'm not sure I understand the purpose of this. We already have something like this that purposefully goes in and extends the data with `None` to make it such every row has the same number of values. Can you clarify the reason behind changing empty strings `""` to `None`? I'd like to know more about your reasoning here

Also if it must be done that way, I think it would be more efficient to do it in the loop starting at line 68. Since we already loop through the data and extend the lists with None type, it doesn't make sense to loop through again after

Reply:
Our reason behind changing the empty strings `""` to `None` was because when trying to create a dataframe using `pandas`, the types of all of the data values have to be homogenous, unless this is a misunderstanding, which could very well be the case! So even with all of the allowed data types that can be accessed using sheetshuttle, we thought this was the best route to ensure that they can be turned into a dataframe without the issue of potentially having empty strings present in a data set that is supposed to be just integers or booleans and so on. 

In our thinking, we also came to a conclusion that with the tools we have to work with currently, there is no way to get around the user having to work around some kind of unwanted data being present in the dataframe that is created. So, we figured to make it easier for the user to deal with this, we opted to make all empty spaces `None` so the user only has to worry about filtering out `None` or `NaN` values. 

Also, in some cases, some of the other approaches we could have gone with, like auto filling empty spaces to zeros when dealing with integers, could lead to issues where the user may want zeros to appear in their google sheet, but then be unable to distinguish between a zero they intended, or a zero that represents an empty space.

Reply:
Ok! That makes a lot of sense. I appreciate all the effort and thought that went into considering the different user cases for this. I'm ok with continuing with this design as long as it gets documented somewhere that the default value for empty cells is `None`. 

With that in mind, I still think that this code fits better with the loop on line 68. What do you think about that?

Lastly, I think we should have some sort of testing to verify that this works as we expect. In this case, doing automated testing is going to be hard so let's stick to few manual testing cases. Can you please set up an example google sheet with missing data and then write a plugin that retrieves the regions in there and tries to print out the date. We probably want to try different data types for the regions as well.

Overall, great work!

Reply:
We will be sure to document that the default value for empty cells is `None`. Do you have any preference where this documentation should be added?

As for the code fitting better with the loop on line 68, I agree with you on this and we'll be sure to make this change once we get the chance.

When it comes to testing, should we then abandon the test we have below? Or should we keep that test, fix it, and make more tests as well?

Reply:
I'm not sure where documenting this fits best. Try to go through the documentations and determine that among your team.  I'm open to you adding wherever you see fit as long as it's clear and describes the behavior accurately.

Sounds good, thanks!

I don't think you should abandon it, however, you should narrow its scope. I'm suggesting two kinds of tests for this PR:
1- Automated tests: This test should be added to either the `test_sheet_collector.py` file or to a new file like `test_util.py`. For these test cases, you should narrow the scope so that only the `fill_to_dimensions()` function gets called. Avoid creating a SheetCollector object and calling the Google Sheets api. Focus on testing only the functionality of `fill_to_dimensions()`.
2- Manual Tests: That's the example I provided in my previous comment about setting up an example google sheet and writing a plugin to get the data from there. After you run that, try to write up a brief report on your findings supported by screenshots and console output and share it with the team.

I hope this clarifies the testing requirement, please let me know if you have additional questions.